StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet.swift

```diff
@@ -153,28 +153,49 @@ public class PaymentSheet {
-                    case .horizontal:
-                        return PaymentSheetViewController(
-                            configuration: self.configuration,
-                            loadResult: loadResult,
-                            analyticsHelper: self.analyticsHelper,
-                            delegate: self
-                        )
-                    case .vertical, .automatic:
-                        let verticalVC = PaymentSheetVerticalViewController(
-                            configuration: self.configuration,
-                            loadResult: loadResult,
-                            isFlowController: false,
-                            analyticsHelper: self.analyticsHelper
-                        )
-                        verticalVC.paymentSheetDelegate = self
-                        return verticalVC
+                let presentPaymentSheet: () -> Void = {
+                    // Set the PaymentSheetViewController as the content of our bottom sheet
+                    let paymentSheetVC: PaymentSheetViewControllerProtocol = {
+                        switch self.configuration.paymentMethodLayout {
+                        case .horizontal:
+                            return PaymentSheetViewController(
+                                configuration: self.configuration,
+                                loadResult: loadResult,
+                                analyticsHelper: self.analyticsHelper,
+                                delegate: self
+                            )
+                        case .vertical, .automatic:
+                            let verticalVC = PaymentSheetVerticalViewController(
+                                configuration: self.configuration,
+                                loadResult: loadResult,
+                                isFlowController: false,
+                                analyticsHelper: self.analyticsHelper
+                            )
+                            verticalVC.paymentSheetDelegate = self
+                            return verticalVC
+                        }
+                    }()
+                    self.bottomSheetViewController.setViewControllers([paymentSheetVC])
+                }
+                if let linkAccount = LinkAccountContext.shared.account, loadResult.elementsSession.shouldShowLink2FABeforePaymentSheet(for: linkAccount, configuration: self.configuration) {
+                    let verificationController = LinkVerificationController(mode: .inlineLogin, linkAccount: linkAccount)
+                    verificationController.present(from: self.bottomSheetViewController) { result in
+                        switch result {
+                        case .completed:
+                            self.presentPayWithNativeLinkController(from: self.bottomSheetViewController, intent: loadResult.intent, elementsSession: loadResult.elementsSession, shouldOfferApplePay: true, shouldFinishOnClose: false) {
```

Comment:
i would have expected shouldOfferApplePay to come from our configuration?

Reply:
Ah yes, good catch!